### DISCUSSION TITLE
Não consegui compilar localmente o projeto

### DISCUSSION THREAD

**Discussion by paulaanasilva** [dgadelha/Portugol-Webstudio#219] [Dúvidas / Perguntas] [answered]
Boa tarde,

Eu baixei o projeto localmente e segui os passos para compilar e executar. Eu teve este erro aqui:

`@portugol-webstudio/ide: > @portugol-webstudio/ide@0.0.0 start
@portugol-webstudio/ide: > ng serve --disable-host-check
@portugol-webstudio/ide: Warning: Running a server with --disable-host-check is a security risk. See https://medium.com/webpack/webpack-dev-server-middleware-security-issues-1489d950874a for more information.
@portugol-webstudio/ide: - Generating browser application bundles (phase: setup)...
@portugol-webstudio/ide: ✔ Browser application bundle generation complete.
@portugol-webstudio/ide: Initial chunk files   | Names         |  Raw size
@portugol-webstudio/ide: vendor.js             | vendor        |   7.49 MB |
@portugol-webstudio/ide: polyfills.js          | polyfills     | 335.99 kB |
@portugol-webstudio/ide: styles.css, styles.js | styles        | 311.65 kB |
@portugol-webstudio/ide: main.js               | main          | 174.87 kB |
@portugol-webstudio/ide: runtime.js            | runtime       |   7.12 kB |
@portugol-webstudio/ide:                       | Initial total |   8.30 MB
@portugol-webstudio/ide: Build at: 2024-05-02T20:45:57.381Z - Hash: 398e124858486878 - Time: 15094ms
@portugol-webstudio/ide: Warning: C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main\Portugol-Webstudio-main\packages\ide\src\app\tab-editor\tab-editor.component.ts depends on 'file-saver'. CommonJS or AMD dependencies can cause optimization bailouts.
@portugol-webstudio/ide: For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
@portugol-webstudio/ide: ./packages/ide/src/app/tab-editor/tab-editor.component.ts:4:0-88 - Error: Module not found: Error: Can't resolve '@portugol-webstudio/runner' in 'C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main\Portugol-Webstudio-main\packages\ide\src\app\tab-editor'
@portugol-webstudio/ide: ./packages/ide/src/app/tab-editor/tab-editor.component.ts:5:0-64 - Error: Module not found: Error: Can't resolve '@portugol-webstudio/runtime' in 'C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main\Portugol-Webstudio-main\packages\ide\src\app\tab-editor'
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:15:60 - error TS2307: Cannot find module '@portugol-webstudio/runner' or its corresponding type declarations.
@portugol-webstudio/ide: 15 import { PortugolExecutor, PortugolWebWorkersRunner } from "@portugol-webstudio/runner";
@portugol-webstudio/ide:                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:16:35 - error TS2307: Cannot find module '@portugol-webstudio/runtime' or its corresponding type declarations.
@portugol-webstudio/ide: 16 import { PortugolJsRuntime } from "@portugol-webstudio/runtime";
@portugol-webstudio/ide:                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:119:13 - error TS7006: Parameter 'event' implicitly has an 'any' type.
@portugol-webstudio/ide: 119       next: event => {
@portugol-webstudio/ide:                 ~~~~~
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:137:14 - error TS7006: Parameter 'error' implicitly has an 'any' type.
@portugol-webstudio/ide: 137       error: error => {
@portugol-webstudio/ide:                  ~~~~~
@portugol-webstudio/ide: ** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **
@portugol-webstudio/ide: × Failed to compile.
@portugol-webstudio/ide: ✔ Browser application bundle generation complete.
@portugol-webstudio/ide: Initial chunk files | Names   | Raw size
@portugol-webstudio/ide: runtime.js          | runtime |  7.12 kB |
@portugol-webstudio/ide: 4 unchanged chunks
@portugol-webstudio/ide: Build at: 2024-05-02T20:46:03.575Z - Hash: 344d9debf966f1b1 - Time: 3890ms
@portugol-webstudio/ide: ./packages/ide/src/app/tab-editor/tab-editor.component.ts:4:0-88 - Error: Module not found: Error: Can't resolve '@portugol-webstudio/runner' in 'C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main\Portugol-Webstudio-main\packages\ide\src\app\tab-editor'
@portugol-webstudio/ide: ./packages/ide/src/app/tab-editor/tab-editor.component.ts:5:0-64 - Error: Module not found: Error: Can't resolve '@portugol-webstudio/runtime' in 'C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main\Portugol-Webstudio-main\packages\ide\src\app\tab-editor'
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:15:60 - error TS2307: Cannot find module '@portugol-webstudio/runner' or its corresponding type declarations.
@portugol-webstudio/ide: 15 import { PortugolExecutor, PortugolWebWorkersRunner } from "@portugol-webstudio/runner";
@portugol-webstudio/ide:                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:16:35 - error TS2307: Cannot find module '@portugol-webstudio/runtime' or its corresponding type declarations.
@portugol-webstudio/ide: 16 import { PortugolJsRuntime } from "@portugol-webstudio/runtime";
@portugol-webstudio/ide:                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:119:13 - error TS7006: Parameter 'event' implicitly has an 'any' type.
@portugol-webstudio/ide: 119       next: event => {
@portugol-webstudio/ide:                 ~~~~~
@portugol-webstudio/ide: Error: src/app/tab-editor/tab-editor.component.ts:137:14 - error TS7006: Parameter 'error' implicitly has an 'any' type.
@portugol-webstudio/ide: 137       error: error => {
@portugol-webstudio/ide:                  ~~~~~
@portugol-webstudio/ide: × Failed to compile.`

Poderia me dar uma orientação sobre como executar localmente ? 
Eu seguir todas as orientações solicitadas:

Executando o código localmente
* Tenho o node instalado, versão v20.9.0

Instalei as dependências do projeto

* npm ci
Compilei os pacotes:
* npm run build
Inicie o servidor de desenvolvimento:
* npm start

**Comment by dgadelha** [comment]
Olá, está me parecendo que o `npm run build` não executou com sucesso. Consegue executar novamente e mandar a saída dele aqui?

**Reply by paulaanasilva** [reply]
Eu imagino que isso aconteceu porque ao executar o primeiro comando "npm ci' apresentou duas vulnerabilidades críticas:

```
# npm audit report

ejs  <3.1.10
Severity: moderate
ejs lacks certain pollution protection - https://github.com/advisories/GHSA-ghr5-ch3p-vcr6
fix available via `npm audit fix`
node_modules/ejs

gh-pages  <5.0.0
Severity: critical
tschaub gh-pages vulnerable to prototype pollution - https://github.com/advisories/GHSA-8mmm-9v2q-x3f9
No fix available
node_modules/gh-pages
  angular-cli-ghpages  *
  Depends on vulnerable versions of gh-pages
  node_modules/angular-cli-ghpages

```

Sabe me ajudar com essas duas ? 


As outras vulnerabilidades identificadas foram essas:

```
tar  <6.2.1
Severity: moderate
Denial of service while parsing a tar file due to lack of folders count validation - https://github.com/advisories/GHSA-f5x3-32g6-xq36
fix available via `npm audit fix`
node_modules/tar
  @lerna/create  7.1.5 - 8.1.2
  Depends on vulnerable versions of tar
  node_modules/@lerna/create
    lerna  6.3.1-beta.0 - 6.3.1-beta.4 || 6.4.2-beta.0 - 8.1.2
    Depends on vulnerable versions of @lerna/create
    Depends on vulnerable versions of tar
    node_modules/lerna

undici  <=5.28.3 || 6.0.0 - 6.11.0
Undici's Proxy-Authorization header not cleared on cross-origin redirect for dispatch, request, stream, pipeline - https://github.com/advisories/GHSA-m4v8-wqvr-p9f7
Undici's Proxy-Authorization header not cleared on cross-origin redirect for dispatch, request, stream, pipeline - https://github.com/advisories/GHSA-m4v8-wqvr-p9f7
Undici's fetch with integrity option is too lax when algorithm is specified but hash value is in incorrect - https://github.com/advisories/GHSA-9qxr-qj54-h672
Undici's fetch with integrity option is too lax when algorithm is specified but hash value is in incorrect - https://github.com/advisories/GHSA-9qxr-qj54-h672
fix available via `npm audit fix`
node_modules/@firebase/auth-compat/node_modules/undici
node_modules/@firebase/auth/node_modules/undici
node_modules/@firebase/firestore/node_modules/undici
node_modules/@firebase/functions/node_modules/undici
node_modules/@firebase/storage/node_modules/undici
node_modules/undici
  @angular-devkit/build-angular  17.1.0-next.0 - 17.3.3 || 18.0.0-next.0 - 18.0.0-next.1
  Depends on vulnerable versions of undici
  Depends on vulnerable versions of vite
  node_modules/@angular-devkit/build-angular
  @firebase/auth  1.5.0-20231112213030 - 1.7.0
  Depends on vulnerable versions of undici
  node_modules/@firebase/auth
    firebase  0.900.22 || 7.9.1-0 - 7.9.1-canary.0396117e || 8.10.0-20217172214 - 8.10.0-canary.f40c0db53 || 9.0.0-20217250818 - 9.0.0-canary.d0d3acb10 || 9.0.1-2021727231341 - 9.0.1-canary.e039e1472 || 9.0.2-2021891633 - 9.0.2-canary.ff9baf70c || 9.0.3-202181503543 - 9.1.0-canary.f7d8324a1 || 9.1.1-2021830195733 - 9.1.1-canary.e70de6201 || 9.1.2-20219523556 - 9.1.2-canary.fc1d36497 || 9.19.1-20230331192943 - 9.19.1-canary.264909862 || 9.22.1-20230524195328 || 10.6.0-20231107192534 - 10.6.0-canary.ebc694a33 || 10.7.0-20231112213030 - 10.10.0
    Depends on vulnerable versions of @firebase/auth
    Depends on vulnerable versions of @firebase/auth-compat
    Depends on vulnerable versions of @firebase/firestore
    Depends on vulnerable versions of @firebase/firestore-compat
    Depends on vulnerable versions of @firebase/functions
    Depends on vulnerable versions of @firebase/functions-compat
    Depends on vulnerable versions of @firebase/storage
    Depends on vulnerable versions of @firebase/storage-compat
    node_modules/firebase
  @firebase/auth-compat  0.4.9-20231107192534 - 0.4.9-canary.a5c1a3594 || 0.5.0-20231112213030 - 0.5.5
  Depends on vulnerable versions of @firebase/auth
  Depends on vulnerable versions of undici
  node_modules/@firebase/auth-compat
  @firebase/firestore  4.3.2-20231027003707 - 4.3.2-canary.ff1a6ec2c || 4.4.0-20231112213030 - 4.5.1
  Depends on vulnerable versions of undici
  node_modules/@firebase/firestore
    @firebase/firestore-compat  <=0.0.900-exp.520ca39d0 || 0.3.22-20231107192534 - 0.3.22-canary.ebc694a33 || 0.3.23-20231112213030 - 0.3.28 || 0.3.30-20240424141009 - 0.3.30-dataconnect-preview.f2ddc3d7b      
    Depends on vulnerable versions of @firebase/firestore
    node_modules/@firebase/firestore-compat
  @firebase/functions  0.11.0-20231112213030 - 0.11.3
  Depends on vulnerable versions of undici
  node_modules/@firebase/functions
    @firebase/functions-compat  0.3.5-20230523183426 - 0.3.5-canary.2a2e2b7ce || 0.3.6-20231112213030 - 0.3.9
    Depends on vulnerable versions of @firebase/functions
    node_modules/@firebase/functions-compat
  @firebase/storage  0.11.2-20230301000120 - 0.11.2-canary.2a2e2b7ce || 0.12.0-20231112213030 - 0.12.3
  Depends on vulnerable versions of undici
  node_modules/@firebase/storage
    @firebase/storage-compat  <=0.0.900-exp.520ca39d0 || 0.1.4-202192711727 - 0.1.4-canary.f27fe4304 || 0.3.2-20230301000120 - 0.3.2-canary.b782bb270 || 0.3.3-20231112213030 - 0.3.6
    Depends on vulnerable versions of @firebase/storage
    node_modules/@firebase/storage-compat

vite  5.1.0 - 5.1.6
Severity: moderate
Vite's `server.fs.deny` did not deny requests for patterns with directories. - https://github.com/advisories/GHSA-8jhw-289h-jh2g
fix available via `npm audit fix`
node_modules/vite

18 vulnerabilities (10 low, 6 moderate, 2 critical)

To address issues that do not require attention, run:
  npm audit fix

Some issues need review, and may require choosing
a different dependency.
```

**Reply by dgadelha** [reply]
Essas vulnerabilidades são apenas um dado informativo e não afetam o funcionamento do código, então não seria isso. Consegue mostrar só a saída do `npm run build`?

**Reply by paulaanasilva** [reply]
Certo, ao executar o 'npm run build' apresentou o seguinte:

```
> portugol-webstudio@0.0.0 build
> lerna run build

lerna notice cli v8.1.2
lerna info Executing command in 6 packages: "npm run build"
@portugol-webstudio/resources: > node --no-warnings --loader ts-node/esm src/atualizar.ts
@portugol-webstudio/antlr: > @portugol-webstudio/antlr@0.0.0 build
@portugol-webstudio/antlr: > tsc
@portugol-webstudio/resources: Remote ETag: W/"fc94c5d444aaa41748ec61d3098825084cc496900321947990d83b3e01265254"
@portugol-webstudio/resources: node:events:492
@portugol-webstudio/resources:       throw er; // Unhandled 'error' event
@portugol-webstudio/resources:       ^
@portugol-webstudio/resources: Error: ENOENT: no such file or directory, open 'C:\C:\Users\ana.silva\Documents\Pessoal\Projeto%20Mestrado\Projeto%20Thiago\Portugol-Webstudio-main_v3\Portugol-Webstudio-main\packages\resources\Portugol-Studio.zip'
@portugol-webstudio/resources: Emitted 'error' event on WriteStream instance at:
@portugol-webstudio/resources:     at emitErrorNT (node:internal/streams/destroy:151:8)
@portugol-webstudio/resources:     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
@portugol-webstudio/resources:     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
@portugol-webstudio/resources:   errno: -4058,
@portugol-webstudio/resources:   code: 'ENOENT',
@portugol-webstudio/resources:   syscall: 'open',
@portugol-webstudio/resources:   path: 'C:\\C:\\Users\\ana.silva\\Documents\\Pessoal\\Projeto%20Mestrado\\Projeto%20Thiago\\Portugol-Webstudio-main_v3\\Portugol-Webstudio-main\\packages\\resources\\Portugol-Studio.zip'
@portugol-webstudio/resources: }
@portugol-webstudio/resources: Node.js v20.9.0
@portugol-webstudio/resources: npm error Lifecycle script `build` failed with error:
@portugol-webstudio/resources: npm error code 1
@portugol-webstudio/resources: npm error path C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main_v3\Portugol-Webstudio-main\packages\resources
@portugol-webstudio/resources: npm error workspace @portugol-webstudio/resources@0.0.0
@portugol-webstudio/resources: npm error location C:\Users\ana.silva\Documents\Pessoal\Projeto Mestrado\Projeto Thiago\Portugol-Webstudio-main_v3\Portugol-Webstudio-main\packages\resources
@portugol-webstudio/resources: npm error command failed
@portugol-webstudio/resources: npm error command C:\WINDOWS\system32\cmd.exe /d /s /c node --no-warnings --loader ts-node/esm src/atualizar.ts
@portugol-webstudio/runtime: > @portugol-webstudio/runtime@0.0.0 build
@portugol-webstudio/runtime: > tsc
@portugol-webstudio/parser: > @portugol-webstudio/parser@0.0.0 build
@portugol-webstudio/parser: > tsc
@portugol-webstudio/runner: > @portugol-webstudio/runner@0.0.0 build
@portugol-webstudio/runner: > tsc
@portugol-webstudio/worker: > @portugol-webstudio/worker@0.0.0 build
@portugol-webstudio/worker: > webpack
@portugol-webstudio/worker: asset worker.js 372 KiB [emitted] [minimized] [big] (name: main)
@portugol-webstudio/worker: orphan modules 1.22 MiB [orphan] 216 modules
@portugol-webstudio/worker: runtime modules 891 bytes 4 modules
@portugol-webstudio/worker: ./src/index.js + 114 modules 892 KiB [built] [code generated]
@portugol-webstudio/worker: WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
@portugol-webstudio/worker: This can impact web performance.
@portugol-webstudio/worker: Assets:
@portugol-webstudio/worker:   worker.js (372 KiB)
@portugol-webstudio/worker: WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
@portugol-webstudio/worker: Entrypoints:
@portugol-webstudio/worker:   main (372 KiB)
@portugol-webstudio/worker:       worker.js
@portugol-webstudio/worker:
@portugol-webstudio/worker: WARNING in webpack performance recommendations: 
@portugol-webstudio/worker: You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
@portugol-webstudio/worker: For more info visit https://webpack.js.org/guides/code-splitting/
@portugol-webstudio/worker: webpack 5.90.3 compiled with 3 warnings in 4406 ms
lerna ERR! npm run build exited 1 in '@portugol-webstudio/resources'
```

**Reply by dgadelha** [reply]
Obrigado, vou dar uma olhada nesse erro do resources aqui

Ao dar `npm start` os outros erros ainda aparecem?

**Reply by dgadelha** [reply, marked as answer]
Corrigi esse erro que estava dando no Windows neste commit aqui: https://github.com/dgadelha/Portugol-Webstudio/commit/0b7e509e7d716b24301d09168f83942b091ae3a4

**Reply by dgadelha** [reply]
Aqui agora está rodando normal no Windows, pode verificar como tá aí?

<img width="1800" alt="image" src="https://github.com/dgadelha/Portugol-Webstudio/assets/10081640/2ef85899-6d68-45c0-b6a1-243fa895317e">

**Reply by paulaanasilva** [reply]
Com a sua última correção, baixei o projeto novamente, executei os comandos e deu tudo certo! Muito obrigada! 

![image](https://github.com/dgadelha/Portugol-Webstudio/assets/38815548/753bd36e-5e06-424c-907c-7d376a513ec0)
